op_math.go

```diff
@@ -362,17 +355,15 @@ func (op elemBinOp) IncrDo(incr Value, inputs ...Value) (err error) {
 	if !op.returnsPtr() {
 		var retVal Value
 		if retVal, err = op.Do(inputs...); err != nil {
-			err = errors.Wrapf(err, doFail, op)
-			return
+			return errors.Wrapf(err, doFail, op)
 		}
 
 		add := newEBOByType(addOpType, incr.Type(), retVal.Type())
 		if retVal, err = add.UnsafeDo(incr, retVal); err != nil {
-			err = errors.Wrapf(err, unsafeDoFail, add)
-			return
+			return errors.Wrapf(err, unsafeDoFail, add)
 		}
-		err = noIncrErr{retVal}
-		return
+		// return errors.New(incrErr) //////////////////// is this ok????
```

Comment:
Nope. Not OK.  IIRC, there are parts which rely on an `error` that has a `Value()` method, for graceful error handling.